waypoint/services/nats_service.py

```diff
@@ -86,44 +117,112 @@ async def process_events(
             topic,
             state,
         )
-
-        subscription = await self._subscribe(
-            group_id=group_id,
-            wallet_id=wallet_id,
-            topic=topic,
-            state=state,
-            look_back=look_back,
-        )
-
-        async def event_generator():
-            end_time = time.time() + duration
-            while not stop_event.is_set():
-                remaining_time = end_time - time.time()
-                logger.trace("remaining_time: {}", remaining_time)
-                if remaining_time <= 0:
-                    logger.debug("Timeout reached")
-                    stop_event.set()
-                    break
-
-                try:
-                    messages = await subscription.fetch(batch=5, timeout=0.2)
-                    for message in messages:
-                        event = orjson.loads(message.data)
-                        yield CloudApiWebhookEventGeneric(**event)
-                        await message.ack()
-                except TimeoutError:
-                    logger.trace("Timeout fetching messages continuing...")
-                    await asyncio.sleep(0.1)
+        # Get the current time in UTC
+        current_time = datetime.now(timezone.utc)
+
+        # Subtract look_back time from the current time
+        look_back_time = current_time - timedelta(seconds=look_back)
+
+        # Format the time in the required format
+        start_time = look_back_time.strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3] + "Z"
+
+        async def event_generator(
+            *,
+            subscription: JetStreamContext.PullSubscription,
+            group_id: str,
+            wallet_id: str,
+            topic: str,
+            state: str,
+            stop_event: asyncio.Event,
+            start_time: str,
```

Comment:
the `start_time` arg will mask the `start_time` that was just defined in line 127

if start_time is never modified again, then it doesn't have to be passed as an arg, as it'll be set in the method definition